browser/search_engines/search_engine_provider_service_browsertest.cc

```diff
@@ -236,6 +245,7 @@ IN_PROC_BROWSER_TEST_F(SearchEngineProviderServiceTest,
       TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO);
   TemplateURL ddg_url(*ddg_data);
 
-  service->SetUserSelectedDefaultSearchProvider(&ddg_url);
+  template_service->SetUserSelectedDefaultSearchProvider(&ddg_url);
+  search_engine_provider_service->OnTemplateURLServiceChanged();
```

Comment:
Same here - seems this is fixing a test which wasn't working properly before. Can you confirm? (if so, this is great!). I'm trying to understand why `OnTemplateURLServiceChanged` call was added

Reply:
thanks for explanation above! 😄